substrate/frame/child-bounties/src/lib.rs

```diff
@@ -797,11 +832,14 @@ impl<T: Config> Pallet<T> {
 	}
 
 	/// The account ID of a child-bounty account.
-	pub fn child_bounty_account_id(id: BountyIndex) -> T::AccountId {
+	pub fn child_bounty_account_id(
+		parent_bounty_id: BountyIndex,
+		child_bounty_id: BountyIndex,
+	) -> T::AccountId {
 		// This function is taken from the parent (bounties) pallet, but the
 		// prefix is changed to have different AccountId when the index of
 		// parent and child is same.
-		T::PalletId::get().into_sub_account_truncating(("cb", id))
+		T::PalletId::get().into_sub_account_truncating(("cb", parent_bounty_id, child_bounty_id))
```

Comment:
Hm, is this truncating in the tests? I saw that you set the AccountId to u128, so it should not, but when i replace it with this, then it fails:

```rust
let sub = ("cb", parent_bounty_id, child_bounty_id);
T::PalletId::get().try_into_sub_account(sub).defensive()
	.unwrap_or_else(|| T::PalletId::get().into_sub_account_truncating(sub))
```

I am always a bit paranoid about these possibly truncatenations, since they behave grossly different between tests and real runtimes and can cause issues.

Reply:
I use u256 in tests and added integrity test